integration/tsconfig-test.ts

```diff
@@ -0,0 +1,125 @@
+import { test, expect } from "@playwright/test";
+import fse from "fs-extra";
+import path from "path";
+import JSON5 from "json5";
+
+import { createFixture, json } from "./helpers/create-fixture";
+
+async function getTsConfig(projectDir: string) {
+  let tsconfigPath = path.join(projectDir, "tsconfig.json");
+  let config = await fse.readFile(tsconfigPath, "utf8");
+  return JSON5.parse(config);
+}
+
+const DEFAULT_CONFIG = {
+  include: ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
+  compilerOptions: {
+    allowJs: true,
+    esModuleInterop: true,
+    forceConsistentCasingInFileNames: true,
+    isolatedModules: true,
+    jsx: "react-jsx",
+    lib: ["DOM", "DOM.Iterable", "ES2019"],
+    moduleResolution: "node",
+    noEmit: true,
+    resolveJsonModule: true,
+    strict: true,
+    target: "es2019",
+    baseUrl: ".",
+    paths: {
+      "~/*": ["./app/*"],
+    },
+  },
+};
+
+// besides baseUrl, due to an upstream issue in tsconfig-paths
+// https://github.com/dividab/tsconfig-paths/pull/208
+test("should output default tsconfig if file is empty", async () => {
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json({ compilerOptions: { baseUrl: "." } }),
+    },
+  });
+
+  let tsconfig = await getTsConfig(fixture.projectDir);
+  expect(tsconfig).toEqual(DEFAULT_CONFIG);
+});
+
+test("should add/update mandatory config", async () => {
+  let fixture = await createFixture({
+    files: {
+      "tsconfig.json": json({
+        ...DEFAULT_CONFIG,
+        compilerOptions: {
+          ...DEFAULT_CONFIG.compilerOptions,
+          isolatedModules: false, // true is required by esbuild
+          // moduleResolution: "node", // this is required by esbuild
```

Comment:
Should this commented-out code be here?

Reply:
i actually wanted to remove the compilerOption alltogether, but that doesn't happen due to the spread, will fix